extensions/podman/src/extension.ts

```diff
@@ -343,6 +350,11 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
   ];
 
   const provider = extensionApi.provider.createProvider(providerOptions);
+
+  // update the status of the provider if the socket is changed, created or deleted
+  disguisedPodmanSocketWatcher = setupDisguisedPodmanSocketWatcher(provider, getSocketPath());
+  extensionContext.subscriptions.push(disguisedPodmanSocketWatcher);
```

Comment:
Unfortunately this isn't working @benoitf any ideas for the implementation? I see we've actually moved "KubeWatcher" to plugin rather than extensions.

I'm able to get it to initialize, but not monitoring.

Reply:
Alternatively, we can implement a `watchSocket` function in `packages/main/src/plugin/provider-impl.ts` ?

Pass in the providers socket and update accordingly.

Reply:
I created https://github.com/cdrage/podman-desktop/pull/2 

the move for k8s is that, as the core is now tracking the kubeconfig file, it sends events to extensions so extensions don't need to track it themselves